Ops/src/OpsWorkerGenetic.cpp

```diff
@@ -147,66 +143,91 @@ void WorkerGenetic::setMutationProbability(double value)
-            mParent1 = i;
-            bestValue = value;
+    std::vector<size_t> sortedIds = getIdsSortedFromWorstToBest();
+
+    // Select parent 1
+    std::vector<size_t> competitors;
+    competitors.push_back(opsRand()*(mNumPoints));
+    competitors.push_back(opsRand()*(mNumPoints));
+    while(competitors.size() < 2) {
+        competitors.push_back(opsRand()*(mNumPoints));
+    }
+    size_t bestCompetitor = 0;
+    for(auto competitor : competitors) {
+        if(competitor > bestCompetitor) {
+            bestCompetitor = competitor;
         }
     }
-
-    bestValue = std::numeric_limits<double>::max();
-    for(size_t i=0; i<mNumPoints; ++i) {
-        double value = mObjectives[i]*opsRand();
-        if(value < bestValue && i != mParent1) {
-            mParent2 = i;
-            bestValue = value;
+    int bestId = sortedIds[bestCompetitor];
+    mParent1 = bestId;
+
+    // Select parent 2
+    competitors.clear();
+    competitors.push_back(opsRand()*(mNumPoints));
+    competitors.push_back(opsRand()*(mNumPoints));
+    while(competitors.size() < mObjectives.size()/3.0) {
+        competitors.push_back(opsRand()*(mNumPoints));
+    }
+    bestCompetitor = 0;
+    for(auto competitor : competitors) {
+        if(competitor > bestCompetitor) {
+            bestCompetitor = competitor;
         }
     }
+    bestId = sortedIds[bestCompetitor];
+    mParent2 = bestId;
 }
 
 
 //! @brief Auxiliary function for perform crossover
 void WorkerGenetic::crossOver()
 {
-    for(size_t i=0; i<mNumParameters; ++i)
+    mCandidatePoints.push_back(mPoints[mParent1]);
+    mCandidatePoints.push_back(mPoints[mParent2]);
+    double doCrossOver = opsRand();
+    if(doCrossOver < mCrossoverProbability)
     {
-        double x1 = mPoints[mParent1][i];
-        double x2 = mPoints[mParent2][i];
 
-        double doCrossOver = opsRand();
-        if(doCrossOver > mCrossoverProbability)
-        {
-            mChild1[i] = x1;
-            mChild2[i] = x2;
-        }
-        else
+        for(size_t i=0; i<mNumParameters; ++i)
         {
+            double x1 = mPoints[mParent1][i];
+            double x2 = mPoints[mParent2][i];
+
             double mean = (x1+x2)/2.0;
-            double dev = fabs(x2-x1)/2.0;
+            double dev = fabs(x2-x1)/10.0;
 
-            mChild1[i] = gaussian(mean,dev,mParameterMin[i],mParameterMax[i]);
-            mChild2[i] = gaussian(mean,dev,mParameterMin[i],mParameterMax[i]);
+            mCandidatePoints[mCandidatePoints.size()-2][i] = gaussian(mean,dev,mParameterMin[i],mParameterMax[i]);
+            mCandidatePoints[mCandidatePoints.size()-1][i] = gaussian(mean,dev,mParameterMin[i],mParameterMax[i]);
         }
     }
-    mCandidatePoints[mParent1] = mChild1;
-    mCandidatePoints[mParent2] = mChild2;
 }
 
 
 //! @brief Auxiliary function for mutating children
 void WorkerGenetic::mutate()
 {
+    std::vector<size_t> eliteIds = getIdsSortedFromWorstToBest();
+    while(eliteIds.size() > mNumElites) {
```

Comment:
Minor issue: OK, I guess this works, but maybe use std::copy instead. To copy the mNumElites last variables, (using iterators)
If you change it, Remember to pre-allocate the destination, or to use std::backinserter

Or maybe use a loop that loops in reverse and copies from "allSortedIds" into eliteIds

Reply:
I changed it to a simpler solution by just creating a new vector in the range `[sortedIds.end()-mNumElites, sortedIds.end()]`. Maybe not the most efficient code, but better than the previous and quite readable I think.

Reply:
I think this solution is the same as what I intended but much simpler